yarn-project/noir-libs/noir-aztec/src/state_vars/public_state.nr

```diff
@@ -1,16 +1,27 @@
+use crate::context::{PrivateContext, PublicContext};
 use crate::oracle::storage::storage_read;
 use crate::oracle::storage::storage_write;
 use crate::types::type_serialisation::TypeSerialisationInterface;
+use dep::std::option::Option;
 
 struct PublicState<T, T_SERIALISED_LEN> {
     storage_slot: Field,
     serialisation_methods: TypeSerialisationInterface<T, T_SERIALISED_LEN>,
 }
 
 impl<T, T_SERIALISED_LEN> PublicState<T, T_SERIALISED_LEN> {
-    fn new(storage_slot: Field, serialisation_methods: TypeSerialisationInterface<T, T_SERIALISED_LEN>) -> Self {
+    fn new(
+        // Note: Passing the contexts to new(...) just to have an interface compatible with a Map.
+        _: Option<&mut PrivateContext>,
+        _: Option<&mut PublicContext>,
```

Comment:
I'm on the fence about this: should we pass both optional contexts to every state var type? Or just the ones they actually need, like in `ImmutableSingleton`? Still, when we introduce traits, this problem will go away.